### DISCUSSION TITLE
Accessibility level of interface and its members

### DISCUSSION THREAD

**Discussion by Wit-2k** [dotnet/csharplang#8581] [General] [answered]
Default accessibility level of an interface is "internal", while interface members are "public" by default. Why?
If I want to implement current interface outside, isn't it that I have to declare this interface with modifier "public"?

**Comment by HaloFour** [comment, marked as answer]
The default accessibility of all types is `internal`.  The default accessibility of members is `private`, but since an interface describes a contract, interface members must be publicly accessible.  Other accessibility levels weren't supported until C# 8.0 with default interface implementation.

**Reply by Wit-2k** [reply]
Seems C# is designed to be this way. I accept and thanks.
